lib/datadog/tracing/contrib/span_attribute_schema.rb

```diff
@@ -22,6 +22,75 @@ def default_span_attribute_schema?
           Datadog.configuration.tracing.span_attribute_schema ==
             Tracing::Configuration::Ext::SpanAttributeSchema::DEFAULT_VERSION
         end
+
+        # implement this function in all target spans/integrations with spankind
+        def set_peer_service(span)
+          should_set_peer_service(span) && set_peer_service_from_source(span)
+          # if above
+          # then remap + remapped from (SKIP)
+          # else
+          # debug that peer service could not be set
+          # end
+        end
+
+        def should_set_peer_service(span)
+          ps = span.get_tag(Tracing::Metadata::Ext::TAG_PEER_SERVICE)
+          if ps && (ps != '')
+            if ps != span.service
+              span.set_tag(Tracing::Metadata::Ext::TAG_PEER_SERVICE_SOURCE, Tracing::Metadata::Ext::TAG_PEER_SERVICE)
+              return false
+            end
+
+            if (ps == span.service) && (span.service != Datadog.configuration.service)
+              span.set_tag(Tracing::Metadata::Ext::TAG_PEER_SERVICE_SOURCE, Tracing::Metadata::Ext::TAG_PEER_SERVICE)
+              return false
+            end
+          end
+
+          if ((span.get_tag(Tracing::Metadata::Ext::TAG_KIND) == Tracing::Metadata::Ext::SpanKind::TAG_CLIENT) ||
+            (span.get_tag(Tracing::Metadata::Ext::TAG_KIND) == Tracing::Metadata::Ext::SpanKind::TAG_PRODUCER)) &&
+              (Datadog.configuration.tracing.span_attribute_schema ==
+                  Tracing::Configuration::Ext::SpanAttributeSchema::VERSION_ONE) # OR if env var is set
+            return true
+          end
+
+          false
+        end
+
+        def set_peer_service_from_source(span)
+          case
+          when span.get_tag(Aws::Ext::TAG_AWS_SERVICE)
+            sources = Array[Aws::Ext::TAG_QUEUE_NAME,
+              Aws::Ext::TAG_TOPIC_NAME,
+              Aws::Ext::TAG_STREAM_NAME,
+              Aws::Ext::TAG_TABLE_NAME,
+              Aws::Ext::TAG_BUCKET_NAME,
+              Aws::Ext::TAG_RULE_NAME,
+              Aws::Ext::TAG_STATE_MACHINE_NAME,]
+          when span.get_tag(Tracing::Contrib::Ext::DB::TAG_SYSTEM)
+            sources = Array[Tracing::Contrib::Ext::DB::TAG_INSTANCE] # DB_NAME tag?
+          when span.get_tag(Tracing::Contrib::Ext::Messaging::TAG_SYSTEM)
+            sources = Array[] # kafka bootstrap servers
```

Comment:
What does this comment mean?
Is it a call for future action? If so, what action?
Is this documenting that `kafka bootstrap servers` have empty sources? If so, we should write that in the comment.

Reply:
These are all markers for tags that do not exist yet but should be added in future PRs as precursors. Will change them to TODO comments